### PR TITLE
Merge re-worked mouse

### PR DESCRIPTION
**Pull request by MattMistele**
Re-worked TranslateMouse, normalizing mouse movements between 0 - 1 before putting them into the controller, and changing the loop to a timer. Although not perfect, this allows for much more precision and smoothness of the mouse. 

Added a help box to help users decide the interval of the mouse timer (ticks).

Also, I mapped the keyboard to different keys that matched the best to what I was playing. If you want to keep the original keys, you can reject my changes to TranslateKeyboard. 

TODO:
Allow for button mapping. Maybe add some presets for the mapping like those optimized for apps or FPS.
Allow the user to change the mouse sensitivity in the app, to offset changing the ticks.
Optimize the mouse code even more to make for a flawless mouse to joystick experience.

### CODE REVIEW

File: XboxKeyboardMouse/MainForm.cs [hunk 1]
```diff
@@ -79,7 +79,7 @@ private void Tickcount_SelectedIndexChanged(object sender, EventArgs e)
         {
             NumericUpDown numeric = (NumericUpDown)sender;
             decimal value = numeric.Value;
-            SetTickCount((int)value);
+            SetTickCount(1);
```
Comment (fqlx) [inline XboxKeyboardMouse/MainForm.cs]: this breaks the saving custom ticks..  why?

File: XboxKeyboardMouse/TranslateKeyboard.cs [hunk 2]
```diff
@@ -21,24 +21,25 @@ class TranslateKeyboard
         private static readonly Dictionary<Key, X360Buttons> buttonsmap = new Dictionary<Key, X360Buttons>
             {
                 { Key.Space, X360Buttons.A },
-                { Key.F, X360Buttons.B},
-                { Key.E, X360Buttons.X },
-                { Key.Q, X360Buttons.Y},
+                { Key.LeftCtrl, X360Buttons.B},
+                { Key.R, X360Buttons.X },
```
Comment (fqlx) [inline XboxKeyboardMouse/TranslateKeyboard.cs]: i'm not against these changing (this is why we need custom config) but every game i've play has 'X' as 'E'

**Comment by fqlx** [comment]
These changes has the mouse running slower on my PC.  I'll take a look at the code this week.

**Comment by MattMistele** [comment]
Interesting. I admit, the way I'm doing I'm running the loop could make it "skip" ticks, something I'm planning to fix in the future, but the original method ran so slow on my computer it was almost unusable, no matter what I set as my ticks. 

A possible reason it might be slow is the loop isn't actually running on the "tick" of the timer, the loop is elsewhere checking to see if the timer has "ticked", leading to a small delay, but it shouldn't be that noticable. Please take a look and let me know what you think.

File: XboxKeyboardMouse/TranslateKeyboard.cs (hunk 2 above)
Comment (MattMistele) [inline XboxKeyboardMouse/TranslateKeyboard.cs]: Agreed. The games I've played had "R" on keyboard and "X" on the controller as reload, which is why the mappable keyboard controls are next on the list.

**Comment by fqlx** [comment]
I don't agree with a lot of these changes but I went ahead and merge it.  The code needs to evolve.

**Comment by MattMistele** [comment]
Ok. I agree that some of the code I wrote needs to change - if its ok with you and you're open, we should chat this labor day weekend sometime about moving forward.

**Pull request merged**